src/Management/src/Endpoint/Utils.cs

```diff
@@ -24,7 +24,7 @@ public static class Utils
     /// <returns>
     /// A file stream with the file's compressed contents.
     /// </returns>
-    public static Stream CompressFile(string filename, string gzFilename, ILogger logger = null)
+    public static Stream CompressFile(string filename, string gzFilename, ILogger logger)
```

Comment:
The null check should be outside the try/catch block, to prevent the catch handler from throwing NullReferenceException while trying to log the original error.